src/libraries/Microsoft.Extensions.Options/gen/Parser.cs

```diff
@@ -439,7 +439,8 @@ private List<ValidatedMember> GetMembersToValidate(ITypeSymbol modelType, bool s
                     // pop the stack
                     _ = _visitedModelTypes.Remove(enumeratedType.WithNullableAnnotation(NullableAnnotation.None));
                 }
-                else if (ConvertTo(attributeType, _symbolHolder.ValidationAttributeSymbol))
+                else if (ConvertTo(attributeType, _symbolHolder.ValidationAttributeSymbol) &&
+                        _compilation.IsSymbolAccessibleWithin(attributeType, _compilation.Assembly))
```

Comment:
I would consider emitting a warning in that case - otherwise users won't know about members that won't be validated

Reply:
Maybe producing `Info` diagnostics more reasonable here.

Reply:
Yeah, I was just going to suggest that. Another way is to check whether `attributeType`'s location is available in source, and if it's the case - we can safely produce a warning.

Reply:
I added the diagnostics.